packages/headless/src/features/commerce/context/cart/cart-analytics-actions.test.ts

```diff
@@ -0,0 +1,207 @@
+    jest.mocked(createRelay).mockReturnValue({
+      emit,
+      getMeta: jest.fn(),
+      on: jest.fn(),
+      off: jest.fn(),
+      updateConfig: jest.fn(),
+      clearStorage: jest.fn(),
+      version: 'foo',
+    });
+  });
+
+  describe('#logCartAction', () => {
+    beforeEach(() => {
+      initEngine();
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    const productWithoutQuantity = {
+      productId: 'product-id-1',
+      name: 'product-name-1',
+      price: 100,
+    };
+
+    const getExpectedCartActionPayload = (
+      action: 'add' | 'remove',
+      quantity: number = 1
+    ) => ({
+      action,
+      product: productWithoutQuantity,
+      quantity,
+      currency: 'USD',
+    });
+
+    it('logs #ec.cartAction with "add" action and correct payload if quantity > 0 and item does not exist in cart', async () => {
+      const preloadedStateWithCartWithoutItems = {
+        ...defaultPreloadedState,
+        cart: {cartItems: [], cart: {}},
+      };
+      initEngine(preloadedStateWithCartWithoutItems);
+
+      engine.dispatch(logCartAction({...productWithoutQuantity, quantity: 1}));
+      await clearMicrotaskQueue();
+
+      expect(emit).toHaveBeenCalledTimes(1);
+      expect(emit).toHaveBeenCalledWith(
+        'ec.cartAction',
+        getExpectedCartActionPayload('add')
+      );
+    });
+
+    it('logs #ec.cartAction with "add" action and correct payload if quantity > 0 and item does not exist in cart and the quantity > 1', async () => {
+      const preloadedStateWithCartWithoutItems = {
+        ...defaultPreloadedState,
+        cart: {cartItems: [], cart: {}},
+      };
+      initEngine(preloadedStateWithCartWithoutItems);
+
+      engine.dispatch(logCartAction({...productWithoutQuantity, quantity: 3}));
+      await clearMicrotaskQueue();
+
+      expect(emit).toHaveBeenCalledTimes(1);
+      expect(emit).toHaveBeenCalledWith(
+        'ec.cartAction',
+        getExpectedCartActionPayload('add', 3)
+      );
+    });
+
+    it('logs #ec.cartAction with "add" action and correct payload if item exists in cart and new quantity > current', async () => {
+      const preloadedStateWithCart = {
+        ...defaultPreloadedState,
+        cart: {
+          cartItems: [],
+          cart: {
+            [productWithoutQuantity.productId]: {
+              ...productWithoutQuantity,
+              quantity: 1,
+            },
+          },
+        },
+      };
+      initEngine(preloadedStateWithCart);
+
+      engine.dispatch(logCartAction({...productWithoutQuantity, quantity: 2}));
+      await clearMicrotaskQueue();
+
+      expect(emit).toHaveBeenCalledTimes(1);
+      expect(emit).toHaveBeenCalledWith(
+        'ec.cartAction',
+        getExpectedCartActionPayload('add')
+      );
+    });
+
+    it('logs #ec.cartAction with "add" action and correct payload if item exists in cart and new quantity > current and the quantity difference is > 1', async () => {
```

Comment:
I'd propose this:
```suggestion
    it('logs #ec.cartAction with "add" action and correct payload if item exists in cart and new quantity > current', async () => {
```
or this:
```suggestion
    it('logs #ec.cartAction with "add" action and correct payload if item exists in cart and new quantity > current and the quantity difference is >= 1', async () => {
```

Reply:
To be sure, is your suggestion to merge two tests? The first one checks the action with a quantity of one, the second checks the action with a quantity above one. We could argue that a single tests could do the trick for both behavior. What do you think?

Reply:
It feels like there's some duplication that could be removed 🤷